security/fma-l1-pectra.md

```diff
@@ -0,0 +1,94 @@
+# Upgrade 12 (L1 Pectra Defense): Failure Modes and Recovery Path Analysis
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Generic Items](#generic-items)
+- [Audit Requirements](#audit-requirements)
+- [Action Items](#action-items)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+|                     |                      |
+| ------------------- | -------------------- |
+| Author              | George Knee          |
+| Created at          | 5th February 2025    |
+| Needs Approval From |                      |
+| Other Reviewers     |                      |
+| Status              | Draft                |
+
+## Introduction
+
+This document is intended to be shared in a public space for reviews and visibility purposes. It covers Upgrade 12, which involves the following changes:
+
+- (Consensus Layer) Ability to parse and validate blocks and receipts with EIP-7702 transactions, as well as blocks with a non-nil EIP-7685 requestsHash field. 
+- (Smart Contracts) For chains using Fault Proofs, L1 contracts updates which reference an updated absolute prestate hash
+
+Each change has its own section below with a list of Failure Modes.
+
+Below are references for this project:
+
+- [Github tracker](https://github.com/orgs/ethereum-optimism/projects/117/views/9)
+- [Specs clarification (Logs)](https://specs.optimism.io/protocol/derivation.html#on-future-proof-transaction-log-derivation)
+- [Specs clarification (L1 Retrieval)](https://specs.optimism.io/protocol/derivation.html#l1-retrieval)
+
+## Failure Modes and Recovery Paths
+
+### FM1: Incidental breaking changes from upstream geth
+
+- **Description:** Incidental changes from merging upstream geth cause unexpected problems. Example: database schema change can break op-geth nodes on rollout.
```

Comment:
```suggestion
- **Description:** Incidental changes from merging upstream geth cause unexpected problems.
```
I'd not mention a db format change or such things as examples. These failures can happen all the time and are not specific to this upgrade, and they are not critical. I'd focus on liveness issues (at time of L1 Pectra activation, because of bugs e.g. in the STF or derivation, not due to failure upgrading nodes) like chain halts or forking nodes and security issues.

Reply:
The liveness issues are covered in the generic FMA document, so no need to replicate them here.

I'd rather include a mention of a specific failure mode we are aware of with this upgrade, so I tweaked this one a little to reference the release notes (which have a good paper trail about the issue). If you feel strongly, we can cut it out.

Reply:
I'm still unsure because it may create unnecessary confusion and discussions. This db corruption problem seems incidental to the L1 Pectra readiness to me. It happened because we update the op-geth dependency, and unfortunately included a lot of breaking changes to the journaling format, not because we made the software aware of the SetCode tx. It feels like we'd then need to publish an FMA every time we update the op-geth dependency.

But I also don't have a strong opinion here. If you or other reviewers feel that it provides value, happy to leave it in.

Reply:
That's a valid point. Definitely agree that it is incidental to the changes which are in the "feature scope" of the upgrade.

Reply:
Removed it!